OneCommunityGlobal/HighestGoodNetworkApp
# Shengwei create setup history modal

# Description
<img width="622" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/70614111-ebda-4540-9863-3170c0186741">


## Related PRS (if any):
This frontend PR is related to the  [767 backend PR](https://github.com/OneCommunityGlobal/HGNRest/pull/767).

## Main changes explained:
- Added file src/components/UserManagement/SetupHistoryPopup.jsx 
- Updated file src/components/UserManagement/UserManagement.jsx to include new SetupHistoryPopup 
- Updated src/components/UserManagement/UserSearchPanel.jsx to include new icon 
- Updated src/components/UserManagement/setupNewUserPopup.jsx to include global state to trigger SetupHistoryPopup  record refresh
- Updated src/utils/URL.js to include new backend API links.


## How to test:
1. check into current branch and [backend PR 767](https://github.com/OneCommunityGlobal/HGNRest/pull/767).
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Setup backend Gmail Oauth and .env in the backend: [Link for video tutorial ](https://docs.google.com/document/d/1HJJTDtWykjGb4IfiIFlHXDfp_IABvmwKxsMPqqF6rG8/edit) 
5. log as admin/owner user
6. go to navigation bar -> Other Links -> User Management
7. Verify the icon exist
<img width="307" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/d5bd06bc-0ef5-40c9-ad17-67304e89ff73"> <br/>
8. Verify that the table is filterable by email, and sortable by Created Date, Expired Date, and Status.
9. Created a new record in the Semd Setup Link modal in user management UI
10. Verify that the Refresh button work:
    - Make sure your backend is setup for email sending feature. This includes
         -  setup Gmail Oauth ([Link](https://docs.google.com/document/d/1HJJTDtWykjGb4IfiIFlHXDfp_IABvmwKxsMPqqF6rG8/edit) )
         -  Change the email variable in the **refreshSetupInvitation** function in **src/controllers/profileInitialSetupController.js** (backend). Comment out `const { email } = result;` and use `const email = 'YOUR_EMAIL_TO_RECEIVE_MSG@gmail.com';`
         - <img width="745" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/a896d40a-6153-4126-95c7-0e6f3fad5923">
    - Click on the Refresh button on a cancelled record:
         - Verify that success toaster message should displayed.
         - Confirm that the record  status changes to Active.
         - Check that the expiration date extends by 3 weeks from the current date.
         - You should receive an email message  <img width="824" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/c7a0b0d7-5a4b-4ceb-a877-eb602a52f15e">
         - The setup link should be valid.  <img width="946" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/d8e71cfe-1a09-4a26-bee6-6ba42a3dd1fd">

     - Click on the Refresh button on a active record: 
         - Verify that success toaster message should displayed.
         - Check that the expiration date extends by 3 weeks from the current date.
         - You should receive an email message <img width="824" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/c7a0b0d7-5a4b-4ceb-a877-eb602a52f15e">
         -  The setup link should be valid.  <img width="946" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/d8e71cfe-1a09-4a26-bee6-6ba42a3dd1fd">
11. Verify that the cancel button is working as expected:
    - Success toaster message should displayed.
    - Record status should change to Cancelled status.
    The setup link should be invalid. You should receive an email with the setup link by hitting the refresh button in the previous step. You should see an invalid message UI once click on the link.  
<img width="863" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/63915a9e-6a81-4985-b203-5235b964556a">


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/6c86eb1e-8d0b-42db-9c15-28b18c81366b



## Note:
You need to Setup gmail oauth feature for testing. Also, change the email variable in the **refreshSetupInvitation** function in **src/controllers/profileInitialSetupController.js** (backend). Comment out `const { email } = result;` and use `const email = 'YOUR_EMAIL_TO_RECEIVE_MSG@gmail.com';` so that the application will send email to the email you defined!
